lib/pages/settings_views/wallet_settings_view/wallet_settings_wallet_settings/lelantus_settings_view.dart

```diff
@@ -105,16 +107,25 @@ class _LelantusSettingsViewState extends ConsumerState<LelantusSettingsView> {
                     height: 20,
                     width: 40,
                     child: DraggableSwitchButton(
-                      isOn: enableLelantusScanning ?? false,
-                      onValueChanged: (newValue) {
+                      isOn: _enableLelantusScanning,
+                      onValueChanged: (newValue) async {
+                        if (_isUpdatingLelantusScanning) return;
+                        setState(() {
+                          _isUpdatingLelantusScanning = true;
```

Comment:
setState not needed here unless the ui needs to be redrawn based on that value being changed